docs/source/best_practices.rst

```diff
@@ -0,0 +1,244 @@
+
+.. code-block:: python
+
+    def intra_partition_sort(df, sort_keys):
+        return df.sort_values(sort_keys)
+
+    c.sql("""
+    SELECT
+        DAYOFMONTH(pickup_datetime) AS dom,
+        HOUR(pickup_datetime) AS hr,
+        *
+    FROM
+        taxi_unsorted
+    DISTRIBUTE BY dom
+    """).map_partitions(intra_partition_sort, ['dom', 'hr']).to_parquet('/data/taxi_sorted')
+
+.. code-block:: python
+
+    sorted_ddf = dd.read_parquet(
+        '/data/taxi_sorted',
+        split_row_groups=False,
+        filters=[
+            [("dom", "==", 15)]
+        ]
+    )
+
+    c.create_table("taxi_sorted", sorted_ddf)
+    .sql("SELECT * FROM taxi_sorted WHERE dom = 15").npartitions
+
+.. code-block::
+
+    1
+
+This comes with a large corresponding boost in computation speed. For example,
+
+.. code-block:: python
+
+    %%time
+    c.sql("SELECT COUNT(*) FROM taxi_unsorted WHERE DAYOFMONTH(pickup_datetime) = 15").compute()
+
+.. code-block::
+
+    CPU times: user 2.4 s, sys: 275 ms, total: 2.68 s
+    Wall time: 2.58 s
+
+.. code-block:: python
+
+    %%time
+    c.sql("SELECT COUNT(*) FROM taxi_sorted WHERE dom = 15").compute()
+
+.. code-block::
+
+    CPU times: user 318 ms, sys: 21.7 ms, total: 340 ms
+    Wall time: 274 ms
+
+
+For a deeper dive into read filtering with Dask, check out `Filtered Reading with RAPIDS & Dask to Optimize ETL <https://medium.com/rapids-ai/filtered-reading-with-rapids-dask-to-optimize-etl-5f1624f4be55>`_.
+
+Avoid Unnecessary Parallelism
+-----------------------------
+
+Additionally, more tasks added to the Dask graph means more overhead added by the scheduler which can be
+a major performance inhibitor at large scales.
+
+For CPUs this isn't as much of an issue, as CPUs tend to have allowance for more workers and CPU tasks tend to take longer, so the additional overhead is relatively less impactful.
+But, for GPUs there's typically only one worker per GPU and tasks tend to be shorter, so the overhead added by a large number of tasks can greatly affect performance.
+
+Improve performance by only creating tasks as necessary. For example, splitting row groups creates more tasks so avoid this if possible.
+
+.. code-block:: python
+
+    weather_dir = '/data/weather_pq_2GB/*.parquet'
+
+
+.. code-block:: sql
+
+    CREATE OR REPLACE TABLE weather_split WITH (
+        location = '{weather_dir}',
+        gpu=True,
+        split_row_groups=True
+    )
+
+.. code-block:: sql
+
+    SELECT COUNT(*) FROM weather_split WHERE type='PRCP'
+
+
+.. code-block:: sql
+
+    CREATE OR REPLACE TABLE weather_nosplit WITH (
+        location = '{weather_dir}',
+        gpu=True,
+        split_row_groups=False
+    )
+
+.. code-block:: sql
+
+    SELECT COUNT(*) FROM weather_nosplit WHERE type='PRCP'
+
+
+Predicate Pushdown
```

Comment:
This section feels like it overlaps a lot with `Sort and Use Read Filtering`. I wonder if we can consolidate these two into a single section?

@randerzander Curious if you have opinions here.

Reply:
Yeah we could consolidate them for now and then split them back out when we're ready to expand the predicate pushdown section.